org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/CompileScope.scala

```diff
@@ -0,0 +1,39 @@
+package org.scalaide.core.internal.project
+
+import java.io.File.separator
+import org.eclipse.core.runtime.IPath
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.resources.IContainer
+
+trait CompileScope {
+  def name: String
+  def dependentScopesInUpstreamProjects: Seq[CompileScope]
+  def isValidSourcePath(projectRelativePath: IPath): Boolean
+  protected def isInPath(that: IPath, pathPatterns: String*): Boolean = {
+    val thatProjectRelative = that.toOSString
```

Comment:
I'm worried about going to OS strings. We need to stay within the Eclipse API in order to support all their features, like linked resources.

`IPath` has a method called `isPrefixOf(path)`, which seems to be what you need here.